src/app/machines/views/MachineList/MachineListTable/MachineCheckbox/MachineCheckbox.tsx

```diff
@@ -47,7 +102,18 @@ const MachineCheckbox = ({
       inputLabel={label}
       isChecked={isChecked ? Checked.Checked : Checked.Unchecked}
       isDisabled={allSelected || groupSelected}
-      onGenerateSelected={(checked) => {
+      onGenerateSelected={(checked, event) => {
+        window.getSelection()?.removeAllRanges();
+        // @ts-ignore shiftKey is usually defined when using click events
+        if (event?.nativeEvent.shiftKey && !groupValue) {
```

Comment:
We're gonna have to narrow this down to MouseEvent or KeyboardEvent type to prevent using @ts-ignore.

Reply:
Could you give more context to this? I'm not sure I follow

Reply:
We know that a nativeEvent here will be of type PointerEvent (which has a `shiftKey`), you can annotate it like below:
```tsx
event: React.ChangeEvent<HTMLInputElement> & {
          nativeEvent: React.PointerEvent;
        }
```

Reply:
oh, understood